### PR TITLE
feat(frontend): util to map transaction icon

### PR DESCRIPTION
**Pull request by AntonioVentilii**
# Motivation

We create an util to help map the Transaction Icon  for all the transaction types. It is used directly in the Transaction component, once expanded to include all the types.

# Changes

- Create util to map the icon, replicating existing logic for ETH, BTC and IC.
- Use zod for the enums involved.
- Expand `Transaction` to include all types.

# Tests

Some tests are provided for the new function.

### CODE REVIEW

**Comment by AntonioVentilii** [comment]
[OISY-130]

File: src/frontend/src/tests/lib/utils/transaction.utils.spec.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			it(`should return IconSend for send and status ${status}`, () => {
+				expect(mapTransactionIcon({ type: 'send', status })).toBe(IconSend);
+			});
+
+			it(`should return IconReceive for receive and status ${status}`, () => {
+				expect(mapTransactionIcon({ type: 'receive', status })).toBe(IconReceive);
+			});
+		});
+
+		allConvertTypes.forEach((type) => {
```
Comment (peterpeterparker) [inline src/frontend/src/tests/lib/utils/transaction.utils.spec.ts:49]: here too you can use a `it.each`

**Review by peterpeterparker — APPROVED** [review]
LGTm, thx

File: src/frontend/src/tests/lib/utils/transaction.utils.spec.ts (hunk 1 above)
Comment (AntonioVentilii) [reply src/frontend/src/tests/lib/utils/transaction.utils.spec.ts]: done

**Comment by AntonioVentilii** [comment]
[OISY-130]

**Pull request merged**